packages/frontend/src/components/EditorSettings/EditorSidebar.tsx

```diff
@@ -0,0 +1,63 @@
+import { Link, useMatch } from 'react-router-dom'
+import { Text } from '@chakra-ui/react'
+import { SidebarContainer, SidebarItem } from '@opengovsg/design-system-react'
+
+import { DrawerLink } from '.'
+
+interface EditorSidebarItemProps {
+  link: DrawerLink
+  closeDrawer: () => void
+}
+
+interface EditorSidebarProps {
+  links: DrawerLink[]
+  closeDrawer: () => void
+}
+
+function EditorSidebarItem({
+  link,
+  closeDrawer,
+}: EditorSidebarItemProps): JSX.Element {
+  const { to, Icon: icon, text } = link
+  const selected = useMatch({ path: to, end: true })
+
+  return (
+    <SidebarItem
+      mx={{ md: '1rem' }}
+      w={{ md: '14.75rem' }}
+      icon={icon}
+      as={Link}
+      to={to}
+      onClick={closeDrawer}
+      isActive={!!selected}
+      color="base.content.default"
+      _hover={{
+        color: 'primary.600',
+        bg: 'interaction.muted.main.hover',
+      }}
+      _active={{
+        color: 'primary.600',
+        bg: 'interaction.muted.main.active',
+      }}
+    >
+      <Text textStyle="subhead-1" ml={4}>
+        {text}
+      </Text>
+    </SidebarItem>
+  )
+}
+
+export default function EditorSidebar(props: EditorSidebarProps) {
```

Comment:
Any reason you didn't re-use the `NavigationSidebar` component from our `Layout` component? Can just refactor and pull that out, instead of duplicating.

Reply:
i think i mentioned to you that currently, the `NavigationSidebar` uses `Context` but over here, I felt it was an overkill to use `Context` so they differ by a bit, will leave this to another linear task to refactor after consulting Stacey on whether we should change the responsive design for flows/apps/executions